rotkehlchen/tests/unit/decoders/test_convex.py

```diff
@@ -0,0 +1,1120 @@
+                    hexstring_to_bytes('0x0000000000000000000000005b186c93a50d3cb435fe2933427d36e6dc688e4b'),  # noqa: E501
+                ],
+            ),
+        ],
+    )
+    dbethtx = DBEthTx(database)
+    dbethtx.add_ethereum_transactions([transaction], relevant_address=None)
+    decoder = EVMTransactionDecoder(
+        database=database,
+        ethereum_manager=ethereum_manager,
+        eth_transactions=eth_transactions,
+        msg_aggregator=msg_aggregator,
+    )
+    events = decoder.decode_transaction(transaction=transaction, tx_receipt=receipt)
+    expected_events = [
+        HistoryBaseEntry(
+            event_identifier=tx_hex,
+            sequence_index=0,
+            timestamp=0,
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=ZERO, usd_value=ZERO),
+            location_label='0x5B186c93A50D3CB435fE2933427d36E6Dc688e4b',
+            notes='Burned 0 ETH in gas from 0x5B186c93A50D3CB435fE2933427d36E6Dc688e4b',
+            counterparty=CPT_GAS,
+            identifier=None,
+            extra_data=None,
+        ), HistoryBaseEntry(
+            event_identifier=tx_hex,
+            sequence_index=343,
+            timestamp=0,
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_CVX,
+            balance=Balance(amount=ONE, usd_value=ZERO),
+            location_label='0x5B186c93A50D3CB435fE2933427d36E6Dc688e4b',
+            notes='Send 1 CVX from 0x5B186c93A50D3CB435fE2933427d36E6Dc688e4b to 0xCF50b810E57Ac33B91dCF525C6ddd9881B139332',  # noqa: E501
+            counterparty=CPT_CONVEX,
+            identifier=None,
+            extra_data=None,
+        ), HistoryBaseEntry(
+            event_identifier=tx_hex,
+            sequence_index=344,
+            timestamp=0,
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.APPROVE,
+            asset=A_CVX,
+            balance=Balance(amount=FVal('1.157920892373161954235709850E+59'), usd_value=ZERO),
+            location_label='0x5B186c93A50D3CB435fE2933427d36E6Dc688e4b',
+            notes='Approve 1.157920892373161954235709850E+59 CVX of 0x5B186c93A50D3CB435fE2933427d36E6Dc688e4b for spending by 0xCF50b810E57Ac33B91dCF525C6ddd9881B139332',  # noqa: E501
+            counterparty='0xCF50b810E57Ac33B91dCF525C6ddd9881B139332',
+            identifier=None,
+            extra_data=None,
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.parametrize('ethereum_accounts', [[string_to_ethereum_address('0x95c5582D781d507A084c9E5f885C77BabACf8EeA')]])  # noqa: E501
+def test_cvx_get_reward(database, ethereum_manager, eth_transactions):
+    msg_aggregator = MessagesAggregator()
```

Comment:
In the tests instead of creating a new instance of the `MessagesAggregator` we could switch to use the `msg_aggregator` from `database`.

Reply:
Well, yes, but as I see from the rest of the codebase a new `MessagesAggregator` is created in almost all tests where it is needed. Am I wrong?

Reply:
okok

Reply:
You haven't done it. One of the reasons to do it is that it doesn't load the websockets messenger here. It is not something we handle in the tests at the moment but reusing them could help in the future